src/cookie.js

```diff
@@ -20,6 +21,45 @@ var reset = function() {
   };
 };
 
+const getHost = (url) => {
+  const a = document.createElement('a');
+  a.href = url;
+  return a.hostname || location.hostname; 
+};
+
+const topDomain = (url) => {
+  const host = getHost(url);
+  const parts = host.split('.');
+  const last = parts[parts.length - 1];
+  const levels = [];
+
+  if (parts.length === 4 && last === parseInt(last, 10)) {
+    return levels;
+  }
+
+  if (parts.length <= 1) {
+    return levels;
+  }
+
+  for (let i = parts.length - 2; i >= 0; --i) {
+    levels.push(parts.slice(i).join('.'));
+  }
+
+  for (let i = 0; i < levels.length; ++i) {
+    const cname = '__tld_test__';
+    const domain = levels[i];
+    const opts = { domain: '.' + domain };
+
+    baseCookie.set(cname, 1, opts);
```

Comment:
my only gripe is that this function seems to be a getter method but it's also setting things into the cookie

Reply:
Yeah. It's ugly. It does clean up after itself though.

There might be a better way of doing this, but I don't want to stray too far from what top-domain's approach without having time to set up proper tests.